hyperbolic/euclid/Arc.py

```diff
@@ -40,6 +40,17 @@ def reverse(self):
     def reversed(self):
         return Arc(self.cx, self.cy, self.r, self.endDeg, self.startDeg,
                    cw=not self.cw)
+    def isPointOnSegment(self, x, y):
+        '''Assumes that Point is on the circle'''
+        px = x-self.cx
+        py = y-self.cy
+        pDeg = math.degrees(math.atan2(py, px))
+        if not self.cw:
+            res = self.startDeg<=pDeg and pDeg<=self.endDeg
+            return res
+        else:
+            res = self.startDeg>=pDeg and pDeg>=self.endDeg
+            return res
```

Comment:
I tried this out and it incorrectly returns False
- when `startDeg > endDeg` and cw=False
- when `startDeg < endDeg` and cw=True
- when the point is on the lower half of the circle (atan2 returns a negative number but startDeg and endDeg are normalized to positive numbers).

Try changing `start`, `end`, and `cw` in this code to test it out:
```python
import drawSvg as draw
from hyperbolic import euclid
import math

start = 75
end = 230
cw = False

x, y, r = 0.2, 0.3, 0.4
circle = euclid.shapes.Circle(x, y, r)
arc = euclid.shapes.Arc(x, y, r, start, end, cw=cw)
print('Drawn shape:', arc)

points = [
    arc.startPoint(),
    arc.endPoint(),
] + [
    (x + r*math.cos(math.radians(deg)),
     y + r*math.sin(math.radians(deg)))
    for deg in range(0, 360, 20)
]

d = draw.Drawing(2.1, 2.1, origin='center')

d.draw(circle, fill='none', stroke_width=0.02, stroke='green')
d.draw(arc, fill='black')
for x, y in points:
    # Blue only if on segment
    fill = 'skyblue' if arc.isPointOnSegment(x, y) else 'red'
    p = euclid.shapes.Circle(x, y, 0.02)
    d.draw(p, fill=fill)

d.setRenderSize(w=400)
d
```

Reply:
I changed the code, and tested it, seems to work in all cases now.
In your code you used isPointOnSegment(self, x, y, tol) is a tolerance a feature needed/wanted? Or is it for testing only?

Reply:
I forgot to mention tolerance.  I noticed some cases where either the start or endpoint was not on the segment because of rounding error.  This would be a good feature (with a default tolerance of something small) but I'd do that in a follow-up PR.